src/LayerPlanBuffer.cpp

```diff
@@ -99,7 +99,8 @@ void LayerPlanBuffer::addConnectingTravelMove(LayerPlan* prev_layer, const Layer
     if (!prev_layer->last_planned_position || *prev_layer->last_planned_position != first_location_new_layer)
     {
         prev_layer->setIsInside(new_layer_destination_state->second);
-        prev_layer->addTravel(first_location_new_layer);
+        const bool force_retract = prev_layer->storage.getSettingBoolean("travel_retract_before_outer_wall") && prev_layer->storage.getSettingBoolean("outer_inset_first");
```

Comment:
It also happens when you have just one wall. I think I'll modify that myself after merging your fix.